crates/notion-core/src/monitor.rs

```diff
@@ -0,0 +1,84 @@
+use std::io::Write;
+use std::process::{Child, Command, Stdio};
+use std::vec::Vec;
+
+use lazycell::LazyCell;
+use serde_json;
+
+use event::Event;
+use notion_fail::Fallible;
+
+pub struct Monitor {
+    monitor_process: Option<Child>,
+}
+
+impl Monitor {
+    /// Returns the current monitor.
+    pub fn new(command: Option<String>) -> Monitor {
+        Monitor {
+            monitor_process: spawn_process(command),
+        }
+    }
+
+    /// send event to the monitor process
+    // if plugin command is not configured, this is a no-op
+    pub fn send_events(&mut self, events: &Vec<Event>) -> () {
+        if let Some(ref mut child_process) = self.monitor_process {
+            let p_stdin = child_process.stdin.as_mut().unwrap();
+
+            let json = serde_json::to_string(&events);
+            match json {
+                Ok(data) => {
+                    write!(p_stdin, "{}", data).expect("Writing data to plugin failed!");
```

Comment:
I think we will want to tighten this `.expect()` up a bit but I've just added it to my [list of warnings](https://paper.dropbox.com/doc/Notion-Warning-Scenarios-vkSds1q3W6LT3C2qVW22M) and we can come back to it later. So don't worry about it for now.

Reply:
ok, I added a note about this